platform/src/main/java/org/hillview/storage/ParquetFileLoader.java

```diff
@@ -127,8 +173,23 @@ private static void appendGroup(
                     break;
                 }
                 case FIXED_LEN_BYTE_ARRAY: {
-                    String s = g.getString(field, 0);
-                    col.append(s);
+                    if (logicalType instanceof IntervalLogicalTypeAnnotation) {
+                        ByteBuffer bb = g.getBinary(field, 0).toByteBuffer();
+                        bb.order(ByteOrder.LITTLE_ENDIAN);
+                        int months = bb.getInt();
+                        int days = bb.getInt();
+                        int milliseconds = bb.getInt();
+                        if (months != 0) {
+                            HillviewLogger.instance.warn("Month is not zero in parquet interval: using conversion of 30 days per month",
+                                    "name: {0}, months: {1}, days: {2}, milliseconds: {3}", cds.get(field).getPath(), months, days, milliseconds);
+                        }
+                        int daysInMonth = 30;
+                        double totalMilliseconds = (days + daysInMonth * months) * Converters.SECONDS_TO_DAY * Converters.MILLIS_TO_SECONDS + milliseconds;
+                        col.append(totalMilliseconds);
+                    } else {
+                        String s = g.getString(field, 0);
```

Comment:
is this always going to work? 
Is this how parquet encodes strings?

Reply:
Added a warning for `BINARY` and `FIXED_LEN_BYTE_ARRAY` columns with missing or unknown logical type annotations. Should help debugging if error does happen during decoding.